### PR TITLE
Document node-role replacement labels

### PR DESCRIPTION
**Pull request by smarterclayton**
Provide documentation for the new alpha labels.

https://github.com/kubernetes/enhancements/issues/1143

### CODE REVIEW

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -303,11 +305,13 @@ Each feature gate is designed for enabling/disabling a specific feature:
   to discover plugins such as [CSI volume drivers](/docs/concepts/storage/volumes/#csi).
 - `KubeletPodResources`: Enable the kubelet's pod resources grpc endpoint.
    See [Support Device Monitoring](https://git.k8s.io/community/keps/sig-node/compute-device-assignment.md) for more details.
+- `LegacyNodeRoleBehavior`: When disabled, legacy behavior in service load balancers and node disruption will ignore the `node-role.kubernetes.io/master` label in favor of the new opt-in labels.
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md]: nit: “new” here feels a bit like a statement about time. Similar to how the docs should [avoid statements about the future](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-statements-about-the-future), I have a mild preference to see a description that will make sense in 5 or 10 releases time.

```suggestion
- `LegacyNodeRoleBehavior`: Enable legacy node filtering behavior for node disruption and for Service load balancers. With this feature gate enabled, Kubernetes treats nodes specially if they are labelled `node-role.kubernetes.io/master`. Also see the feature gates `NodeDisruptionExclusion` and `ServiceNodeExclusion`.
```

It sounds to me as if there should be a page somewhere in the docs that explains `node-role.kubernetes.io` labels and whether these are intended to be meaningful to cluster operators (yes), to controllers (no), or both (well, no).
Can log an issue if that'd help?

Comment (smarterclayton) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md]: Yeah, an issue would be good.

Comment (smarterclayton) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md]: Updated, let me know if this makes sense

**Comment by smarterclayton** [comment]
Any other comments?

**Pull request merged**